// This file is part of Net Failover Manager.
//
// Net Failover Manager is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, either version 3 of the License, or
// (at your option) any later version.
//
// Net Failover Manager is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with Net Failover Manager.  If not, see
// <https://www.gnu.org/licenses/>.

syntax = "proto3";

package net_failover_manager;

service NetworkConfig {
  rpc GetDefaultGw(DefaultGwRequest) returns (DefaultGwResponse) {}
  rpc GetIfStatus(IfStatusRequest) returns (IfStatusResponse) {}
  rpc ForceNewGateway(ForceNewGatewayRequest)
      returns (ForceNewGatewayResponse) {}
}

message DefaultGwRequest {}

message DefaultGwResponse {
  string default_gw_interface = 1;
  // next available id = 2
}

message IfStatusRequest {}

message IfStatus {
  string if_name = 1;
  string status = 2;
  string last_checked_at = 3;
  // next available id = 4.
}

message IfStatusResponse {
  repeated IfStatus interface_status = 1;
  // next available id = 2
}

message ForceNewGatewayRequest {
  string if_name = 1;
  // next available id = 2.
}
message ForceNewGatewayResponse {}
